[O5fNSvsN] Цены

AlexeyFrunza opened the pull request:
-- добавлен виджет PriceList
-- добавлен компонент PriceCard для отображения карточек в PriceList виджете
-- реализован функционал переключения списка услуг по togglebar'у (UTabs)

Commit:
price-card component

Commit:
[O5fNSvsN] Цены
-- добавлен price-card компонент

Commit:
[O5fNSvsN] Цены
- добавил компонент карточки для прайс листа
- реализовал фичу переключения списка прайс листа через таб

Commit:
[O5fNSvsN] Цены
- добавил компонент карточки для прайс листа
- реализовал фичу переключения списка прайс листа через таб

Commit:
Merge branch 'feature/O5fNSvsN' of https://github.com/weekkkk/x-install--landing-page into feature/O5fNSvsN

Commit:
[O5fNSvsN] Цены
-- добавлены градиентные фигуры по бокам блока "Цены"

weekkkk commented on src/widgets/price-list/ui/index.vue:
```diff
@@ -0,0 +1,177 @@
+<script setup lang="ts">
+import type { UiPriceCardProps } from "~/src/shared/ui/price-card";
```
не нужно импортить его, сделай файл `index.ts` внутри папки `shared/ui` и из него экспорт пропсов для карточки

weekkkk commented on src/widgets/price-list/ui/index.vue:
странный нейминг слайса, лучше назвать слайс `price`, а внутри него сделать компонент `list`
т.е. `widgets/price/ui/list/index.vue`

weekkkk commented on src/widgets/price-list/ui/index.vue:
```diff
@@ -0,0 +1,177 @@
+<script setup lang="ts">
+import type { UiPriceCardProps } from "~/src/shared/ui/price-card";
+
+const items = [
+  {
+    label: "IOS",
+  },
+  {
+    label: "Android",
+  },
+];
+
+const iosServices: UiPriceCardProps[] = [
+  {
+    service: `Мотивированные\nустановки в AppStore`,
+    serviceAccent: `Мотивированные\nустановки`,
+    serviceDescription: `Выведение вашего приложения в топ выдачи по ключевым словам. Стоимость указана\nза одну мотивированную установку по одному\nключевому слову`,
```
такие вещи лучше выносить в папку `consts` внутри виджета, раскидывать по файлам и забирать из `consts/index.ts`

weekkkk commented on src/widgets/price-list/ui/index.vue:
```diff
@@ -0,0 +1,177 @@
+    ],
+  },
+];
+
+const androidServices: UiPriceCardProps[] = [
+  {
+    service: `Мотивированные\nустановки в Play Market`,
+    serviceAccent: `Мотивированные\nустановки`,
+    serviceDescription: `Выведение вашего приложения в топ выдачи по ключевым словам. Стоимость указана\nза одну мотивированную установку по одному\nключевому слову`,
+    guarantees: [
+      "Любое количество установок",
```
аналогично

weekkkk commented on src/widgets/price-list/ui/index.vue:
```diff
@@ -0,0 +1,177 @@
+      v-model="activeTab"
+    />
+    <div class="flex flex-col gap-y-4">
+      <UiPriceCard
+        v-for="item in currentComponents"
+        :service="item.service"
+        :service-accent="item.serviceAccent"
+        :service-description="item.serviceDescription"
+        :guarantees="item.guarantees"
+        :price="item.price"
+        :is-unit-price="item.isUnitPrice"
+      />
```
`:key` передавать всегда нужно, когда `v-for` используешь, если кто-то увидит сразу поймет, что ты не сечешь
https://vuejs.org/guide/essentials/list#maintaining-state-with-key

Commit:
[O5fNSvsN] Цены
-- добавлен экспорт UiPriceCardProps в shared/ui
-- все статичные виджета PriceList данные были пересены в price/ui/list/consts/..
-- добавлен ключ для перебора UiPriceCard

AlexeyFrunza reviewed (commented):
[O5fNSvsN] Цены 
-- правки по замечаниям
